x-pack/plugin/ml/qa/basic-multi-node/src/javaRestTest/java/org/elasticsearch/xpack/ml/integration/MlRescorerIT.java

```diff
@@ -0,0 +1,273 @@
+                                    "default_left": true,
+                                    "left_child": 1,
+                                    "right_child": 2
+                                  },
+                                  {
+                                    "node_index": 1,
+                                    "leaf_value": 1.0
+                                  },
+                                  {
+                                    "node_index": 2,
+                                    "leaf_value": 12.0
+                                  }
+                                  ],
+                                  "target_type": "regression"
+                                }
+                              }
+                              ]
+                            }
+                          }
+                        }
+                      }""");
+        createIndex(INDEX_NAME, Settings.builder().put("number_of_shards", randomIntBetween(1, 3)).build(), """
+            "properties":{
+             "product":{"type": "keyword"},
+             "cost":{"type": "integer"}}""");
+        indexData("{ \"product\": \"TV\", \"cost\": 300 }");
+        indexData("{ \"product\": \"TV\", \"cost\": 400 }");
+        indexData("{ \"product\": \"VCR\", \"cost\": 150 }");
+        indexData("{ \"product\": \"VCR\", \"cost\": 180 }");
+        indexData("{ \"product\": \"Laptop\", \"cost\": 15000 }");
+        refreshAllIndices();
+    }
+
+    @SuppressWarnings("unchecked")
+    public void testLtrSimple() throws Exception {
+        Response searchResponse = search("""
+            {
+            "query": {
+              "match": { "product": { "query": "TV"}}
+            },
+            "rescore": {
+                    "window_size": 10,
+                    "inference": {
+                        "model_id": "basic-ltr-model"
+                        }
+                }
+
+            }""");
+
+        Map<String, Object> response = responseAsMap(searchResponse);
+        assertThat((List<Double>) XContentMapValues.extractValue("hits.hits._score", response), contains(17.0, 17.0));
+    }
+
+    @SuppressWarnings("unchecked")
+    public void testLtrSimpleDFS() throws Exception {
+        Response searchResponse = searchDfs("""
+            {
+            "query": {
+              "match": { "product": { "query": "TV"}}
+            },
+            "rescore": {
+                    "window_size": 10,
+                    "inference": {
+                        "model_id": "basic-ltr-model"
+                        }
+                }
+
+            }""");
+
+        Map<String, Object> response = responseAsMap(searchResponse);
+        assertThat(response.toString(), (List<Double>) XContentMapValues.extractValue("hits.hits._score", response), contains(17.0, 17.0));
+    }
+
+    @SuppressWarnings("unchecked")
+    public void testLtrSimpleEmpty() throws Exception {
+        Response searchResponse = search("""
```

Comment:
What are we testing in the empty hits tests? That assertions are not tripped (should we put a comment for that)?  Otherwise it looks like obvious tests.

Reply:
@mayya-sharipova yes, making sure we are not doing weird work when its unnecessary. I can remove these, but I had them as a way to ensure my stuff was working as I was changing things internally.

Reply:
I think it is great to keep them, may be with extra comments.